sdmetrics/single_table/new_row_synthesis.py

```diff
@@ -108,9 +108,12 @@ def compute_breakdown(cls, real_data, synthetic_data, metadata=None,
 
                 row_filter.append(field_filter)
 
+            engine = None
+            if len(row_filter) >= 32:  # Limit set by NPY_MAXARGS
+                engine = 'python'
             try:
-                matches = real_data.query(' and '.join(row_filter))
-            except TypeError:
+                matches = real_data.query(' and '.join(row_filter), engine=engine)
+            except (TypeError, ValueError):
                 if len(real_data) > 10000:
                     warnings.warn('Unable to optimize query. For better formance, set the '
```

Comment:
@frances-h if you're running into the `ValueError` due to >32 columns, then we don't need to show the warning.

I think the warning is only for `TypeError`, whose root cause is [a different issue](https://github.com/pandas-dev/pandas/issues/22435).

Reply:
Good point, we actually don't need to catch the `ValueError` anymore since we won't try to use `numexpr` if there's too many arguments.